erasure-readfile.go

```diff
@@ -32,6 +32,7 @@ func (e erasure) ReadFile(volume, path string, startOffset int64) (io.ReadCloser
 		return nil, errInvalidArgument
 	}
 
+	remaining := totalSize
 	var wg = &sync.WaitGroup{}
```

Comment:
Look at the current master and branch that implementation as its more cleaner which will handle all the cases here. The current code was written to avoid taking totalSize. 

The problem is that padding causes issues and we cannot do anything about that.